applications/ISO20022-Implementation-POC.md

```diff
@@ -0,0 +1,476 @@
+1. **Account management for banks:** Banks create an account and transfer some DOT to it.
+2. **Transaction management:** Banks can submit a hashed (and salted) ISO20022 message along with the receiving bank account and DOT amount to the pallet in order to transfer a certain amount of DOT to the receiving bank.
+3. **Payment reversal mechanism:** Banks will be able to revert payments in order to be compliant with ISO20022 messages such as pain.007.001.11 (CustomerPaymentReversalV11) or pacs.007.001.12 (FIToFIPaymentReversalV12).
+
+##### 2. Client for Banks
+In order for banks to use Polkadot as a payment settlement system, they will need to run a client which offers the following funcionality:
+
+1. API to receive customer payment instructions
+2. ISO20022 message parser/creator
+3. Send extrinsics to pallet for transfer
+4. Send extrinsic to pallet for reversal
+5. Send/Read ISO messages to bus
+6. Verify relevant ISO messages on chain
+7. Client account management system
+    - Create for each client an account number
+    - Store for each client a list of hashes form the chain
+    - Functionality to derive client balance from onchain hashes
+8. Hashmap-like storage of all ISO messages and salts
+
+We will use NodeJs to build this MVP client.
+
+##### 3. Interbank Message Bus
+As the blockchain only stores hashes of the ISO 20022 messages in order to preserve privacy, we need some a  communication system to exchange ISO20022 messages between banks.
+
+
+#### Payment Flow
+The below diagram describes the steps necessary to complete an ISO20022 payment using the Substrate pallet as the settlement system. Customer1 with a bank account at Bank1 is sending DOT to Customer2 with an account at bank Bank2. In this example we use two ISO20022 messages (for an example please see the appendix):
+1. pain.001.001.11 (CustomerCreditTransferInitiationV11) for the customer to initiate the payment.
+2. pacs.008.001.11 (FIToFICustomerCreditTransferV11) for the interbank transfer.
+
+![Payment Flow](https://solid-bit.com/w3f/iso20022.png "Payment Flow")
+
+#### Design Choices
+
+1. We do not process ISO20022 messages onchain in order to preserve privacy.
+2. We do not use offchain-workers for message processing also for privacy reasons.
+3. We do not create customer accounts on chain, because they could easily be linked to a real person, again breaking privacy.
+4. We do not use one centralized message processing institution, as we want to give each bank full access to their on-chain funds, at least guaranteeing decentralization on the interbank level.
+
+
+
+
+### Ecosystem Fit
+
+Many competing blockchain ecosystems are already ISO20022 compliant (see above). If Polkadot as well wants to become compatible with the traditional finance industry, an ISO20022 integration will be the right place to start.
```

Comment:
Can you give some examples which blockchains are ISO20022 compatible and how they achieved that?

Reply:
Blockchains that claim to be ISO20022 compliant are for example Ripple and Cardano. How they actually achieve this is pretty vague in my eyes. For example Ripple writes in their [ISO20022 one pager](https://ripple.com/lp/iso-overview/):

"To help enable this next step in global interoperability and meet the evolving needs of our customers, Ripple is now part of the ISO 20022 Standards Body-the first member focused on Distributed Ledger Technology (DLT)."

What this means on a technical level does not seem very transparent. The way I see it, it is not even totally clear what it means for a blockchain to be ISO20022 compliant, as ISO20022 is a messaging protocol intended to simplify and standardize communication about financial transactions  between trusted entities.

So, I see this proposal as an experiment to bring more clarity to this question by building an ISO20022 integration that is simple and sound and can be used and tested in practice.